ingestion/src/metadata/ingestion/source/messaging/kinesis/metadata.py

```diff
@@ -114,55 +141,96 @@ def yield_topic(
         return 1000000
 
-    def _get_topic_details(self, topic_name: str) -> Dict[str, Any]:
+    def _get_topic_details(self, topic_name: str) -> Optional[KinesisSummaryModel]:
         try:
-            topic = self.kinesis.describe_stream_summary(StreamName=topic_name)
-            return topic["StreamDescriptionSummary"]
+            topic_summary = self.kinesis.describe_stream_summary(StreamName=topic_name)
+            return KinesisSummaryModel(**topic_summary)
         except Exception as err:
             logger.debug(traceback.format_exc())
             logger.warning(
                 f"Error while fetching topic partitions for topic: {topic_name} - {err}"
             )
-        return {}
+        return None
 
     def _get_topic_partitions(self, topic_name: str) -> List[str]:
-        all_partitions, has_more_partitions, args = [], True, {"StreamName": topic_name}
+        all_partitions, has_more_partitions, args = (
+            [],
+            True,
+            KinesisStreamArgs(StreamName=topic_name),
+        )
         try:
             while has_more_partitions:
-                partitions = self.kinesis.list_shards(**args)
+                partitions = self.kinesis.list_shards(**args.dict())
+                kinesis_partitions_model = KinesisPartitions(**partitions)
                 all_partitions.extend(
-                    [part["ShardId"] for part in partitions["Shards"]]
+                    [
+                        partition.ShardId
+                        for partition in kinesis_partitions_model.Shards or []
+                    ]
                 )
-                has_more_partitions = partitions.get("NextToken")
-                args["NextToken"] = partitions.get("NextToken")
+                has_more_partitions = kinesis_partitions_model.NextToken
+                args.NextToken = has_more_partitions
         except Exception as err:
             logger.debug(traceback.format_exc())
             logger.warning(
                 f"Error while fetching topic partitions for topic: {topic_name} - {err}"
             )
         return all_partitions
 
+    def yield_topic_sample_data(
+        self, topic_details: BrokerTopicDetails
+    ) -> TopicSampleData:
+        """
+        Method to Get Sample Data of Messaging Entity
+        """
+        try:
+            if self.context.topic and self.generate_sample_data:
+                yield OMetaTopicSampleData(
+                    topic=self.context.topic,
+                    sample_data=self._get_sample_data(
+                        topic_details.topic_name,
+                        topic_details.topic_metadata.partitions,
+                    ),
+                )
+        except Exception as err:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Error while yielding topic sample data for topic: {topic_details.topic_name} - {err}"
```

Comment:
Hey @OnkarVO7, can you resolve this
<img width="607" alt="Screenshot 2023-03-23 at 11 49 34 AM" src="https://user-images.githubusercontent.com/40225091/227120155-d02e45b5-d296-4d20-81f2-15910ccc8907.png">

Reply:
code smell

Reply:
handled the code smell